rlp/codec.py

```diff
@@ -218,7 +222,6 @@ def decode(rlp, sedes=None, strict=True, **kwargs):
         obj = sedes.deserialize(item, **kwargs)
         if hasattr(obj, '_cached_rlp'):
             obj._cached_rlp = rlp
-            assert not isinstance(obj, Serializable) or not obj.is_mutable()
```

Comment:
This was removed in favor of having mutable `Serializable` instances automatically invalidate their cache.